openmc/statepoint.py

```diff
@@ -92,13 +91,8 @@ class StatePoint(object):
     """
 
     def __init__(self, filename):
-        if filename.endswith('.h5'):
-            import h5py
-            self._f = h5py.File(filename, 'r')
-            self._hdf5 = True
-        else:
-            self._f = open(filename, 'rb')
-            self._hdf5 = False
+        import h5py
```

Comment:
On second thought, the Python API (minus ``openmc.statepoint``) could still be useful for input generation even if a user doesn't need to analyze HDF5 statepoint and summary files (*e.g.*, only the eigenvalue in the text output is needed). It's up to you whether you think ``h5py`` should be made a mandatory dependency by moving the import to the top of the file.